src/com/owncloud/android/authentication/AuthenticatorAsyncTask.java

```diff
@@ -72,18 +72,32 @@ protected RemoteOperationResult doInBackground(Object... params) {
             );
             result = operation.execute(client);
 
+            String permanentLocation = null;
             if (operation.wasRedirected()) {
                 RedirectionPath redirectionPath = operation.getRedirectionPath();
-                String permanentLocation = redirectionPath.getLastPermanentLocation();
-                result.setLastPermanentLocation(permanentLocation);
+                permanentLocation = redirectionPath.getLastPermanentLocation();
             }
 
             // Operation - get display name
             if (result.isSuccess()) {
-                GetRemoteUserInfoOperation remoteUserNameOperation = new GetRemoteUserInfoOperation();
-                result = remoteUserNameOperation.execute(client);
+                if (permanentLocation == null) {
```

Comment:
A bit more compact and DRY version of the code can be achieved by reference aliasing, which allows to remove one of the conditions and leave it as a single if:

```
if (result.isSuccess()) {
    GetRemoteUserInfoOperation remoteUserNameOperation = new GetRemoteUserInfoOperation();
    if (permanentLocation != null) {
        // we can't assume that any subpath of the domain is correctly redirected; ugly stuff
        OwnCloudClient redirectedClient = OwnCloudClientFactory.createOwnCloudClient(
            Uri.parse(AccountUtils.trimWebdavSuffix(permanentLocation)),
            mContext,
            true
        );
        redirectedClient.setCredentials(credentials);
        client = redirectedClient;
    }
    result = remoteUserNameOperation.execute(client);
}
```